packages/@pollyjs/adapter-puppeteer/src/index.js

```diff
@@ -191,47 +183,29 @@ export default class PuppeteerAdapter extends Adapter {
 
   async passthroughRequest(pollyRequest) {
     const { page } = this.options;
-    const { id, order, url, method, body } = pollyRequest;
+    const { id, order, url, method, headers, body } = pollyRequest;
     const requestId = `${this.polly.recordingId}:${id}:${order}`;
-    const headers = {
-      ...pollyRequest.headers,
-      [PASSTHROUGH_REQ_ID_HEADER]: requestId
-    };
+    const parsedUrl = new URL(url, true);
+
+    parsedUrl.query[PASSTHROUGH_REQ_ID_QP] = requestId;
 
     try {
       const response = await new Promise((resolve, reject) => {
         this[PASSTHROUGH_PROMISES].set(requestId, { resolve, reject });
 
         // This gets evaluated within the browser's context, meaning that
         // this fetch call executes from within the browser.
-        page.evaluate((url, options) => fetch(url, options), url, {
-          method,
-          headers,
-          body
-        });
+        page.evaluate(
+          (url, options) => fetch(url, options),
+          parsedUrl.toString(),
+          { method, headers, body }
+        );
       });
-      let responseBody;
-
-      try {
-        responseBody = await response.text();
-      } catch (error) {
-        /*
```

Comment:
This is no longer needed since im resolving the promise on `requestfinished` instead of on `response`. That's where it should have been resolving in the first place.